index.js

```diff
@@ -1,31 +1,100 @@
 const instance_skel = require('../../instance_skel');
+var Client  = require('node-rest-client').Client;
 
 class instance extends instance_skel {
 	constructor(system, id, config) {
 		super(system, id, config);
 
-		this.actions(); // export actions
+		this.waiting = false;
+		this.connectionID = 0;
+		this.authenticated = false;
+		this.authToken = "";
+		this.availableClips = [];
+
+		this.state = {
+			'TransportState': 0
+		}
+
+		this.t_states = {
+			UNKNOWN:"Unknown",
+			IDLE:"Idle",
+			RECORDING:"Recording",
+			FORWARD:"Forward",
+			FORWARD_2X:"Forward 2X",
+			FORWARD_4X:"Forward 4X",
+			FORWARD_8X:"Forward 8X",
+			FORWARD_16X:"Forward 16X",
+			FORWARD_32X:"Forward 32X",
+			FORWARD_STEP:"Forward Step",
+			REVERSE:"Reverse",
+			REVERSE_2X:"Reverse 2X",
+			REVERSE_4X:"Reverse 4X",
+			REVERSE_8X:"Reverse 8X",
+			REVERSE_16X:"Reverse 16X",
+			REVERSE_32X:"Reverse 32X",
+			REVERSE_STEP:"Reverse Step",
+			PAUSED:"Paused",
+			IDLE_ERROR:"Idle Error",
+			RECORD_ERROR:"Record Error",
+			PLAY_ERROR:"Play Error",
+			PAUSE_ERROR:"Pause Error",
+			SHUTDOWN:"Shutdown"
+		}
 
-		// Example: When this script was committed, a fix needed to be made
-		// this will only be run if you had an instance of an older "version" before.
-		// "version" is calculated out from how many upgradescripts your intance config has run.
-		// So just add a addUpgradeScript when you commit a breaking change to the config, that fixes
-		// the config.
+		this.actions(); // export actions
 
-		this.addUpgradeScript(function () {
+		this.addUpgradeScript(function (config) {
 			// just an example
-			if (self.config.host !== undefined) {
-				self.config.old_host = self.config.host;
+			if (config.host !== undefined) {
+				config.old_host = config.host;
 			}
 		});
+
+		this.addUpgradeScript(function (config) {
+			if (config.polling == undefined) {
+				config.polling = false; //Default polling off for existing users
+			}
+
+			if (config.password == undefined){
+				config.password = ""; //Default password to be empty
+			}
+			this.saveConfig();
```

Comment:
`this.saveConfig();` should not be called here.  This should actually look something for like:
```javascript
		this.addUpgradeScript(function (config) {
			let changed = false;

			if (config.polling == undefined) {
				config.polling = false; //Default polling off for existing users
				changed = true;
			}

			if (config.password == undefined){
				config.password = ""; //Default password to be empty
				changed = true;
			}

			return changed;
		});

Reply:
Also, should `config.pollingRate` be initialized with its default?

Reply:
Is there some documentation related to upgrade scripts someplace? I had to stumble my way through that as I could only find one other module with an upgrade script via google.

Reply:
Haha ... no, and its been on my list to write it for like a year

Reply:
I just pushed a commit that implements your requests